### ISSUE TITLE
Pebble support for handling YAML data structures (function, filters)

### ISSUE THREAD

**Issue by kriko** [kestra-io/kestra#3258]
### Feature description

With the introduction of [TemplatedTask](https://github.com/kestra-io/kestra/pull/3191) and for typical GitOps engineering tasks, having the possibility to work with YAML inside Kestra tasks would sure come in handy.

Feature proposal, similarily to current behaviour, introduce support for YAML.

### `yaml()` function
Similarily to `json()` [function](https://kestra.io/docs/concepts/expression/function#json) [[code](https://github.com/kestra-io/kestra/blob/develop/core/src/main/java/io/kestra/core/runners/pebble/functions/JsonFunction.java)] support the deserialization / objectification of YAML strings to variables.

### `| yaml` filter
Similarily to `| json` [filter](https://kestra.io/docs/concepts/expression/filter/json#json) [[code](https://github.com/kestra-io/kestra/blob/develop/core/src/main/java/io/kestra/core/runners/pebble/filters/JsonFilter.java)] support the serialization / construction of YAML strings from variables.

### `| yq` filter 
Similarily to `| jq` [filter](https://kestra.io/docs/concepts/expression/filter/json#jq) [[code](https://github.com/kestra-io/kestra/blob/develop/core/src/main/java/io/kestra/core/runners/pebble/filters/JqFilter.java)] enable the use of yq library, which essentially is a mapper for yaml -> json to jq. So there are multiple implementation options.

_Optional workaround would be to parse yaml to an object and then use jq._ Eg. `yaml(inputs.yaml_object) | jq`

### `| indent(n)` string filter
With the use of YAML comes the _issue_ of indentation - especially when constructing YAML from multiple objects separately or having to nest generated YAML within an existing templated structure. Thus I propose a simple filter to apply indentation to strings, that would add `n` number of spaces before each line (except for the first one).

Even better, `indent(n, prefix=" ")`, where `n` is the amount of multiplication of `prefix` that is added before each line in a newline separated string.

NB! Important to note, especially given the examples below in the comment - the first line should *NOT* be indented - this is because with most use-cases we already indent the first line.

**Comment by kriko** [comment]
To better illustrate the feature proposal, especially with TemplatedTask and the usage of Kubernetes PodCreate - here is a sample flow, provided runtime configuration and expected task definition that will be run.

### templated-task - flow definition
```
id: templated-task
namespace: kestra.test

labels:
  my-label: "will-be-sent-to-k8s"
  other-label: "value"

inputs:
  # the following represents the object resources with nested structure as supported by Kestra
  # thus they are accessible under the inputs.resources object
  - name: resources.limits.cpu
    type: STRING
    defaults: "1"
    required: true

  - name: resources.limits.memory
    type: STRING
    defaults: "1000Mi"
    required: true

  - name: resources.requests.cpu
    type: STRING
    required: false # note that this is optional and when not filled, it should be skipped in the definition

  - name: resources.requests.memory
    type: STRING
    required: false # note that this is optional and when not filled, it should be skipped in the definition

# other sample using variables, since they allow to explicitly define lists in yaml
variables:
  env: # is presented as a list of objects
    - name: "var_string"
      value: "VALUE"
    - name: "var_int"
      value: 123

tasks:
  - id: dynamic
    type: io.kestra.core.tasks.templating.TemplatedTask
    spec: |
      id: "{{ flow.namespace | slugify }}-{{ flow.id | slugify }}"
      type: io.kestra.plugin.kubernetes.PodCreate
      namespace: kestra
      metadata:
        labels:
          {{ labels | yaml | indent(4) }}
      spec:
        containers:
        - name: debug
          image: alpine:latest
          env:
            {{ variables.env | yaml | indent(6) }}
          command:
            - 'bash'
            - '-c'
            - 'printenv'
        resources:
          {{ inputs.resources | yaml | indent(4) }}
```

### Flow execution context (provided inputs, variables etc.)
A sample run configuration, expressed in YAML with all execution parameters present.
```
flow:
  id: templated-task
  namespace: kestra.test

labels:
  my-label: "will-be-sent-to-k8s"
  other-label: "value"

inputs:
  resources.limits.cpu: "2"
  resources.limits.memory: "1000Mi"
  resources.requests.cpu: "1"
  # resources.requests.memory: # not providing

# other sample using variables, since they allow to explicitly define lists in yaml
variables:
  env: # is presented as a list of objects
    - name: "var_string"
      value: "VALUE"
    - name: "var_int"
      value: 123
```

### Resulting generated task
The following task would be generated and run by TemplatedTask, with the parameters given above.
```
id: "kestra-test-templated-task"
type: io.kestra.plugin.kubernetes.PodCreate
namespace: kestra
metadata:
  labels:
    my-label: "will-be-sent-to-k8s"
    other-label: "value"
spec:
  containers:
  - name: debug
    image: alpine:latest
    env:
      - name: "var_string"
        value: "VALUE"
      - name: "var_int"
        value: 123
    command:
      - 'bash'
      - '-c'
      - 'printenv'
  resources:
    limits:
      cpu: "2"
      memory: "1000Mi"
    requests:
      cpu: "1"
```

**Comment by kriko** [comment]
Can be closed when PR #3283 and PR #3284 are merged to develop.
As for the propsed | yq filter, decided not to implement it, since the used Jackson library doesn't incldue YQ and it can easily be mitigated by the following:

```
yaml(yaml_object) | jq(..)
# or
yaml(yaml_object) | jq(..) | yaml # to print out yaml
```
